src/Routing/HookMenuRoutes.php

```diff
@@ -79,74 +84,148 @@ private function convertToRoute(string $module, string $path, array $definition)
                 $parameters[$placeholder] = $parameter;
                 $pathParts[] = '{' . $placeholder . '}';
             }
+            ++$key;
         }
+
         foreach ($loadArguments as &$loadArgument) {
-            $loadArgument = match (true) {
-                is_int($loadArgument) => $pathParts[$loadArgument],
-                $loadArgument === 'map' => $pathParts,
-                default => $loadArgument,
-            };
+            switch ($loadArgument) {
+                case 'map':
+                    $loadArgument = &$pathParts;
+                    break;
+
+                default:
+                    $loadArgument = $this->ensureArgument($loadArgument, $pathParts, $parameters);
+            }
         }
         $pageArguments = $definition['page arguments'];
         foreach ($pageArguments as &$pageArgument) {
-            if (is_int($pageArgument)) {
-                $pageArgument = $pathParts[$pageArgument];
-            }
+            $pageArgument = $this->ensureArgument($pageArgument, $pathParts, $parameters);
+        }
+        $titleArguments = $definition['title arguments'];
+        foreach ($titleArguments as &$titleArgument) {
+            $titleArgument = $this->ensureArgument($titleArgument, $pathParts, $parameters);
+        }
+        $accessArguments = $definition['access arguments'];
+        foreach ($accessArguments as &$accessArgument) {
+            $accessArgument = $this->ensureArgument($accessArgument, $pathParts, $parameters);
         }
+
+        $defaults = [];
+        $pageCallback = match ($definition['page callback']) {
+            'drupal_get_form' => array_shift($pageArguments),
+            default => $definition['page callback'],
+        };
         if (isset($definition['file'])) {
             $filePath = $definition['file path'] ?: $this->moduleHandler->getModule($module)->getPath();
             $definition['include file'] = $filePath . '/' . $definition['file'];
             if (file_exists($definition['include file'])) {
                 require_once $definition['include file'];
             }
         }
-        $route = new Route('/' . implode('/', $pathParts));
-        $route->setDefault('_title', $definition['title']);
+        if (is_callable($pageCallback)) {
+            $skip = $definition['page callback'] === 'drupal_get_form' ? 2 : 0;
+            $reflectedPageCallback = match (true) {
+                is_array($pageCallback) => new \ReflectionMethod(...$pageCallback),
+                strpos($pageCallback, '::') !== false => new \ReflectionMethod(
+                    ...explode('::', $pageCallback, 2)
+                ),
+                default => new \ReflectionFunction($pageCallback),
+            };
```

Comment:
This is similar to `\Drupal\Component\Utility\ArgumentsResolver::getReflector` but that's a protected method.  And `getArguments` does some of the same steps with extracting parameters. I didn't find anything in Symfony easily, but maybe we can create a helper class for resolving the parameters.

Or maybe that's overkill because we'll only use it here. What if we create a `getResolver` and `extractParameters` method that this calls. Then its able to be extracted if needed.